src/StreamJsonRpc.Shared/Reflection/JsonRpcMethodAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have a different name.  If applied on a method, the method's CLR name can no longer be used for remote invocation.
+    /// </summary>
+    /// <remarks>
+    /// This attribute should be used when rpc message method names can be different from the actual CLR method names.
+    /// Useful in cases where rpc message method names contain illegal characters for CLR method names, i.e. "text/OnDocumentChanged".
+    /// 
+    /// If methods are overloaded, each overload must define its own <see cref="JsonRpcMethodAttribute"/>  with all the same values.  
+    /// Conflicts will result in error being thrown during <see cref="JsonRpc"/> construction.
```

Comment:
I'm glad we expect conflicts to be thrown up-front. But why then do the docs on the two dictionary fields we now have on JsonRpc state that their presence is for enforcement? Is that stale?